wire-library/wire-runtime-swift/src/main/swift/CodingUserInfoKey+Wire.swift

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2023 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+extension ProtoDecoder {
+    /// The Decoding strategy to use for ProtoEnum types
+    /// Defaults to .throwError
+    public typealias CodableEnumDecodingStrategy = UnknownEnumValueDecodingStrategy
+}
+
+extension ProtoEncoder {
+    /// The encoding strategy to use for ProtoEnum types
+    /// Defaults to .string
+    public enum CodableEnumEncodingStrategy {
+        case string
+        case integer
+    }
+}
+
+extension CodingUserInfoKey {
+    /// Control the encoding of ProtoEnum types
```

Comment:
I would prefer not to use `JSONEncoder` since _technically_ any Encoder can use this.
That said, I do like making them properties (and debated doing that before)

Reply:
_gah_ I can't make it settable on all (en/de)coders... OK JSON namespacing it is

Reply:
OK I rooted it off of the relevant JSON coder, but kept the user info keys public.